packages/apollo-link-error/src/index.ts

```diff
@@ -45,7 +45,7 @@ export const onError = (errorHandler: ErrorHandler): ApolloLink => {
                 forward,
               });
 
-              if (retriedResult) {
+              if (typeof retriedResult !== 'undefined' && typeof retriedResult.subscribe === 'function') {
```

Comment:
Wouldn't it be better to just write retriedResult && typeof retriedResult.subscribe? This can still error when retriedResult is null